hazelcast/src/test/java/com/hazelcast/internal/util/JavaVersionTest.java

```diff
@@ -0,0 +1,77 @@
+package com.hazelcast.internal.util;
+
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+
+import static com.hazelcast.internal.util.JavaVersion.JAVA_1_6;
+import static com.hazelcast.internal.util.JavaVersion.JAVA_1_7;
+import static com.hazelcast.internal.util.JavaVersion.JAVA_1_8;
+import static com.hazelcast.internal.util.JavaVersion.JAVA_1_9;
+import static com.hazelcast.internal.util.JavaVersion.UNKNOWN;
+import static org.junit.Assert.*;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class JavaVersionTest extends HazelcastTestSupport {
+
+    @Test
+    public void parseVersion() throws NoSuchFieldException, IllegalAccessException {
+        assertEquals(JAVA_1_6, JavaVersion.parseVersion("1.6"));
+        assertEquals(JAVA_1_7, JavaVersion.parseVersion("1.7"));
+        assertEquals(JAVA_1_8, JavaVersion.parseVersion("1.8"));
+        assertEquals(JAVA_1_9, JavaVersion.parseVersion("9-ea"));
+        assertEquals(JAVA_1_9, JavaVersion.parseVersion("9"));
+        assertEquals(UNKNOWN, JavaVersion.parseVersion("foo"));
+    }
+
+    @Test
+    public void testIsAtLeast_1_6() {
+        assertTrue(JavaVersion.isAtLeast(JAVA_1_6, JAVA_1_6));
+        assertFalse(JavaVersion.isAtLeast(JAVA_1_6, JAVA_1_7));
+        assertFalse(JavaVersion.isAtLeast(JAVA_1_6, JAVA_1_8));
+        assertFalse(JavaVersion.isAtLeast(JAVA_1_6, JAVA_1_9));
+        assertTrue(JavaVersion.isAtLeast(JAVA_1_6, UNKNOWN));
```

Comment:
If you move all `UNKNOWN` tests to the beginning of the methods, they will be easier to read, since you will have just a single `assertTrue()` to `assertFalse()` transition, not forth and back.

If you do so, please also align `parseVersion()`, so all tests have the same order.